### PR TITLE
Fix milu parameters

### PR DESCRIPTION
**Pull request by blattms**
- --m-ilu-variant did not make any sense (Parameter system does not correctly treat consecutive uppercase letters)
- improved help strings.

### CODE REVIEW

File: opm/autodiff/NewtonIterationBlackoilInterleaved.hpp [hunk 1]
```diff
@@ -122,9 +122,9 @@ namespace Opm
             EWOMS_REGISTER_PARAM(TypeTag, int, LinearSolverRestart, "The number of iterations after which GMRES is restarted");
             EWOMS_REGISTER_PARAM(TypeTag, int, FlowLinearSolverVerbosity, "The verbosity level of the linear solver (0: off, 2: all)");
             EWOMS_REGISTER_PARAM(TypeTag, int, IluFillinLevel, "The fill-in level of the linear solver's ILU preconditioner");
-            EWOMS_REGISTER_PARAM(TypeTag, std::string, MIluVariant, "Specify which variant of the modified-ILU preconditioner ought to be used");
+            EWOMS_REGISTER_PARAM(TypeTag, std::string, MiluVariant, "Specify which variant of the modified-ILU preconditioner ought to be used. Possible variants are: ILU (default, plain ILU), MILU_1 (lump diagonal with dropped row entries), MILU_2 (lump diagonal with the sum of the absolute values of the dropped row  entries), MILU_3 (if diagonal is positive add sum of dropped row entrires. Otherwise substract them), MILU_4 (if diagonal is positive add sum of dropped row entrires. Otherwise do nothing");
```
Comment (andlaus) [inline opm/autodiff/NewtonIterationBlackoilInterleaved.hpp]: this is going to be a looong line in the usage message.

**Comment by andlaus** [comment]
looks good. I'll merge later today if nobody beats me to it and nobody vetos the new "monster help messages" for the parameters.

File: opm/autodiff/NewtonIterationBlackoilInterleaved.hpp (hunk 1 above)
Comment (blattms) [reply opm/autodiff/NewtonIterationBlackoilInterleaved.hpp]: Then please tell me how to break it or better: point to the documentation of the command line parameter system.

Comment (andlaus) [reply opm/autodiff/NewtonIterationBlackoilInterleaved.hpp]: regarding the documentation of the parameter system: the whole API is these four macros:

https://github.com/OPM/ewoms/blob/master/ewoms/common/parametersystem.hh#L59-L119

if you want to write some prose on how they are used within eWoms and/or flow you're most welcome to do so.

Comment (blattms) [reply opm/autodiff/NewtonIterationBlackoilInterleaved.hpp]: > https://github.com/OPM/ewoms/blob/master/ewoms/common/parametersystem.hh#L59-L119

I just took another look. But I still do not see how that explains my problems with the command line parameters or how to properly use them. I did not even use those macros in my code.

**Comment by blattms** [comment]
BTW I see that there is no way how the command line parameter system can detect the correct usage if conecutive upper case letters would be allowed to form one word.

File: opm/autodiff/NewtonIterationBlackoilInterleaved.hpp (hunk 1 above)
Comment (andlaus) [reply opm/autodiff/NewtonIterationBlackoilInterleaved.hpp]: > "Read the source, dude" is not documentation.

is "run Doxygen, dude" better? ;)

> Concerning the help page: If I do not get a list of possible values for a parameter, then the help is rather useless. BTW there are more parameters lacking this.

true. I've been thinking for a while that having proper support for "enumeration parameters" would be nice. It just never was a priority for me yet.

also, something like git's approach to help would be -- err -- helpful: print a as-brief-as-possible usage message for all parameters on `--help` and output all the gory on `--help $paramname`. adding this functionality and the necessary docstrings would be a medium-sized project, though.

> I just took another look. But I still do not see how that explains my problems with the command line parameters or how to properly use it.

I suppose that your issues are more related to how this machinery is used than the machinery itself. I don't have a good answer for that, maybe the best thing to do is to step through the parameter registration with a debugger.

**Comment by andlaus** [comment]
> BTW I see that there is no way how the command line parameter system can detect the correct usage if conecutive upper case letters would be allowed to form one word.

in the interpretation of the camelCase convention used here, uppercase letters denote word boundaries, i.e.`ISTLSolver` translates to "i s t l solver", not "ISTL solver".

**Comment by andlaus** [comment]
seems like there were no vetos. merging, thanks for the patch.

**Comment by OPMUSER** [comment]
If you email me the help text for the command line parameters in a plain text file, doesn't have to be clean, as long as I can understand the information, then I'll incorporate it into the manual in the next release. 

What I would like to do in the future is to provide some guidance in the manual on how to use these parameters and how they can be used to attempt to resolve various computational problems.  I can do this for Eclipse, but not OPM as the numerical controls and the numerical schemes are completely different. If anybody has any input on this just sent me the information in a plain text file and I'll' sort out the formatting etc. Figures can be in the text file or separate if necessary. It would also be good to have some examples showing the problem and the fix that solves the issue.

Note most convergence problems are normally related to input data issues and I can certainly write a few sections on what to check and how resolve these type of problems.

**Pull request merged**